src/serverenvironment.cpp

```diff
@@ -1306,6 +1303,20 @@ void ServerEnvironment::step(float dtime)
 				obj->m_messages_out.pop();
 			}
 		}
+
+		// calculate a simple moving average
+		m_avg_ao_time = m_avg_ao_time * 0.9f + timer.stop(true) * 0.1f;
+		// allow up to 10% of the budget interval
+		const u32 max_time_ms = m_cache_ao_interval * 1000 / 10;
```

Comment:
better to keep float here

Reply:
Why is that? You're worried about the resolution when m_cache_ao_interval is very small?

Reply:
no, i just want to use a float instead of u32 for the comparison with the interval